app/models/order_cycle.rb

```diff
@@ -150,6 +153,21 @@ def self.earliest_closing_times
     ]
   end
 
+  def attachable_payment_methods
+    Spree::PaymentMethod.available(:both).
+      joins("INNER JOIN distributors_payment_methods
+             ON payment_method_id = spree_payment_methods.id").
+      where("distributor_id IN (?)", distributor_ids).
+      distinct
+  end
+
+  def attachable_shipping_methods
+    Spree::ShippingMethod.frontend.
+      joins(:distributor_shipping_methods).
+      where("distributor_id IN (?)", distributor_ids).
+      distinct
+  end
```

Comment:
If deleted shipping methods shouldn't be included in the query then we need to change this to simply `joins(:selected_shipping_methods)`.

Reply:
The `attachable_shipping_methods` method should return a list of shipping method options that are allowed to be attached rather than ones that are actually attached which is `selected_shipping_methods`.

Reply:
Ah, sorry, I was reading this too quickly. Now I see what you are doing here. :+1: